water/include/water/Dialect/Wave/IR/WaveAttrs.td

```diff
@@ -306,6 +306,40 @@ def DeviceConstraintAttr : AttrDef<WaveDialect, "DeviceConstraint"> {
 // Other attributes
 //-----------------------------------------------------------------------------
 
+def WaveIndexSymbolEnum
+    : I32EnumAttr<"WaveIndexSymbol", "Index symbol", [
+  I32EnumAttrCase<"DEVICE_DIM_0", 0, "DD0">,
+  I32EnumAttrCase<"DEVICE_DIM_1", 1, "DD1">,
+  I32EnumAttrCase<"DEVICE_DIM_2", 2, "DD2">,
+  I32EnumAttrCase<"WORKGROUP_0", 3, "WG0">,
+  I32EnumAttrCase<"WORKGROUP_1", 4, "WG1">,
+  I32EnumAttrCase<"WORKGROUP_2", 5, "WG2">,
+  I32EnumAttrCase<"THREAD_0", 6, "T0">,
+  I32EnumAttrCase<"THREAD_1", 7, "T1">,
+  I32EnumAttrCase<"THREAD_2", 8, "T2">,
+  I32EnumAttrCase<"GPR_NUMBER", 9, "GPR_NUM">,
+]> {
+  let cppNamespace = "::wave";
+  let genSpecializedAttr = 0;
+}
+
+def WaveIndexSymbolAttr : EnumAttr<WaveDialect, WaveIndexSymbolEnum, "WaveIndexSymbol"> {
+  let mnemonic = "index_symbol";
+  let assemblyFormat = "`<` $value `>`";
+  let description = [{
+    Specialized symbol used in index expressions in the Wave dialect.
+  }];
+
+  let extraClassDeclaration = [{
+    static constexpr std::initializer_list<::llvm::StringRef> AllCases = { }]# !interleave(!foreach(case, enum.enumerants, "\"" # case.str # "\""), ", ") #[{ };
```

Comment:
Nice tablegen trick! Please add a commment above explaining what it does though :)